pkg/driver/controller.go

```diff
@@ -99,8 +100,35 @@ func (d *controllerService) CreateVolume(ctx context.Context, req *csi.CreateVol
 		}
 	}
 
-	volumeParams := req.GetParameters()
-	fsType := volumeParams[FsTypeKey]
+	var (
+		fsType      string
+		volumeType  string
+		iopsPerGB   int
+		isEncrypted bool
+		kmsKeyId    string
+	)
+
+	for key, value := range req.GetParameters() {
+		switch strings.ToLower(key) {
+		case FsTypeKey:
+			fsType = value
+		case VolumeTypeKey:
+			volumeType = value
+		case IopsPerGBKey:
+			iopsPerGB, err = strconv.Atoi(value)
+			if err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "Could not parse invalid iopsPerGB: %v", err)
+			}
+		case EncryptedKey:
+			if value == "true" {
+				isEncrypted = true
+			} else {
+				isEncrypted = false
+			}
+		case KmsKeyIdKey:
+			kmsKeyId = value
+		}
```

Comment:
Shouldn't we explicitly default to an `codes.InvalidArgument` error if we get something unexpected?

Do you know what the in-tree plugin does in this case? If it simply ignores the parameter, we may do the same.

Reply:
It does return errors: https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/awsebs/aws_util.go#L180

Updated